go/database/mpt/archive_trie.go

```diff
@@ -205,7 +205,7 @@ func (a *ArchiveTrie) GetCode(block uint64, account common.Address) (code []byte
 	return a.head.GetCodeForHash(info.CodeHash), nil
 }
 
-func (a *ArchiveTrie) GetCodes() (map[common.Hash][]byte, error) {
+func (a *ArchiveTrie) GetCodes() map[common.Hash][]byte {
```

Comment:
The error actually cannot be created this and it easifies the code inside `immutable` and. I would like to include this but I can create first PR with this change, merge it into main and then use it here if you feel like its necessary...

Reply:
ok, lets keep this change, since this is not an API method I think it does not matter much if it is changed or not. I just wanted to stress that it is always better to have only related changes in PRs.